packages/namekit-react/src/components/Identity.tsx

```diff
@@ -0,0 +1,292 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  ReactNode,
+} from "react";
+import {
+  createClient,
+  Network,
+  type SecurePrimaryNameResult,
+} from "@namehash/nameguard";
+
+interface IdentityContextType {
+  network: string;
+  address: string;
+  returnNameGuardReport: boolean;
+  loadingState: "loading" | "error" | "success";
+  error?: string;
+  identityData?: SecurePrimaryNameResult;
+  followersCount?: string;
+}
+
+const IdentityContext = createContext<IdentityContextType | null>(null);
+
+const useIdentity = () => {
+  const context = useContext(IdentityContext);
+
+  if (!context) {
+    throw new Error("useIdentity must be used within an IdentityProvider");
+  }
+
+  return context;
+};
+
+interface SubComponentProps {
+  className?: string;
+  children?: ReactNode;
+}
+
+interface RootProps {
+  address: string;
+  network?: Network;
+  className?: string;
+  children: ReactNode;
+  returnNameGuardReport?: boolean;
+}
+
+const Root = ({
+  address,
+  network = "mainnet",
+  className,
+  children,
+  returnNameGuardReport = false,
+  ...props
+}: RootProps) => {
+  const [data, setData] = useState<IdentityContextType>({
+    address,
+    network,
+    returnNameGuardReport,
+    loadingState: "loading",
+  });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setData((prev) => ({ ...prev, loadingState: "loading" }));
+
+        const nameguard = createClient({ network });
+
+        const result = await nameguard.getSecurePrimaryName(address, {
```

Comment:
Suggest that we add ReactQuery as a dependency / peer dependency for NameKit React.

The web3 community has generally adopted ReactQuery as an unofficial standard. There's a ton of various libraries and packages already using it or built on it.

For example, strongly suggest having a look at the following:

1. wagmi. https://wagmi.sh/ The same team that built wagmi built viem https://viem.sh/. Pretty much everyone is using both libraries in their web3 apps. viem is a more low-level library that's suitable for use on a backend or frontend. wagmi is a frontend library that wraps viem. wagmi uses ReactQuery for all their network requests.
2. onchainkit. https://onchainkit.xyz/ This is built as a layer on top of wagmi.
3. rainbowkit. https://www.rainbowkit.com/ This is also built as a layer on top of wagmi.
4. Etc..

In fact, suggest that we not only add ReactQuery as a dependency, but we also add wagmi as a dependency. We shouldn't try to reinvent the whole universe. Pretty much everyone who might use NameKit React is also going to be using wagmi, so we better make it convenient for everything to work together.